homeassistant/components/mystrom/__init__.py

```diff
@@ -22,6 +22,24 @@
 _LOGGER = logging.getLogger(__name__)
 
 
+async def _async_get_device_state(
+    device: MyStromSwitch | MyStromBulb, ip_address: str
+) -> None:
+    try:
+        await device.get_state()
+    except MyStromConnectionError as err:
+        _LOGGER.error("No route to myStrom plug: %s", ip_address)
+        raise ConfigEntryNotReady() from err
+
+
+def _get_mystrom_bulb(host: str, mac: str) -> MyStromBulb:
+    return MyStromBulb(host, mac)
+
+
+def _get_mystrom_switch(host: str) -> MyStromSwitch:
+    return MyStromSwitch(host)
```

Comment:
I currently use this method to be able to inject mocks in the tests. Any other method would be preferred!

Reply:
why do you need to make it a function instead of calling constructor right away like it was ?

Reply:
Because I want to inject a mock in the tests and I could not figure out how to patch a constructor call. (the mock is now used to mock the behaviour of the external library better, since the current tests arent representing this well)